src/agent/oneplatformcontroller.ts

```diff
@@ -0,0 +1,332 @@
+// Copyright 2022 Google LLC
```

Comment:
nit: I would probably use hyphens for these file names:

```
oneplatform-controller, firebase-controller
```

Reply:
I like it.  Done.